docs/recommendations.md

```diff
@@ -0,0 +1,64 @@
+Invocation flow for a recommendations function annotated in [`Common.RecommendationsFunction`](../vocabularies/Common.md#RecommendationsFunction)
+
+```mermaid
+sequenceDiagram
+  actor client as Fiori client
+  box ABAP
+    participant server as OData service
+    participant task as New task
+  end
+  participant AI as BTP recommendations service
+  activate client
+  note over client: sticky session
+  client ->>+ server: POST $batch
+  server ->> server: PATCH SalesOrder('A')
+  par
+    server ->> server: GET SalesOrder('A')?$expand=... for side effects
+  and
+    server ->> server: GET SalesOrder('A')/ns.Recommendations?Targets=...<br>Prefer: respond-async
+    server ->>+ task: call function ... starting new task
+  end
+  server -->>- client: $batch response with<br>Location header for recommendations
+  task ->>+ AI: compute recommendations
+  client ->>+ server: GET «Location header for recommendations»
+  activate client
+  note right of client: outside sticky session
```

Comment:
It is only necessary to be outside of the session if the call waits for the results. On the client this would, however, mean that we introduce some special mechanism to do this polling call outside of the session. This would not be very nice.

Reply:
But doesn't repeated calling into the session make things slower overall, even if each individual request is rather short?

Reply:
Beside the time the requests block the session? I hope not.